subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/SubscriptionsManager.kt

```diff
@@ -239,25 +245,81 @@ class RealSubscriptionsManager @Inject constructor(
         _hasSubscription.emit(false)
     }
 
-    private suspend fun checkPurchase() {
+    private suspend fun checkPurchase(
+        packageName: String,
+        purchaseToken: String,
+    ) {
+        var retryCompleted = false
+
+        suspend fun retry(
```

Comment:
@marcosholgado not sure if this is what you had in mind https://app.asana.com/0/0/1206571662770261/f so I just tagged https://app.asana.com/0/0/1206498134325580/f for this task instead.